c_emulator/riscv_sim.c

```diff
@@ -1075,3 +1109,56 @@ int main(int argc, char **argv)
   flush_logs();
   close_logs();
 }
+
+bool rv_cfg_c_ext_enable(char * isa_str, char * ext_pattern)
+{
+  pcre2_code          *re;
+  int                 errornumber;
+  PCRE2_SIZE          erroroffset;
+  pcre2_match_data    *match_data;
+  int                 rc;
+
+  re = pcre2_compile(
+    ext_pattern,            /* the pattern */
```

Comment:
Currently all users are using this as a glorified strstr(3). But that's broken; it'll match any substring, not just a full extension name. That is, calling it with rv64izcbom and m will return true, making it look like M is enabled, for example, as will rv64izfhmin with zfh. As it stands this function is totally broken. Besides, regular expressions are not a good tool for parsing ISA strings; you really want to write a custom parser as it's far easier to write, understand, debug and alter in future.